openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/taxa.sql

```diff
@@ -3,51 +3,165 @@
+            as ancestry, --exclude 'Life' from ancestry
+        left(coalesce(
+            catalog_of_life.name_english,
+            manual_name_additions.vernacular_name, --name_manual,
+            taxa.name -- name_inaturalist,
+        ), 5000) as title,
+        (case when catalog_of_life.name_english is not null or
+            manual_name_additions.vernacular_name is not null
+            then to_jsonb(array_fill(cast((taxa.name, 'inaturalist') as openverse_tag), array[1]))
+            end) inaturalist_name_tag,
+        tags_nonenglish_vernacular
+    from inaturalist.taxa
+    LEFT JOIN catalog_of_life
+        on (cast(md5(taxa.name) as uuid) = catalog_of_life.md5_scientificname)
+    LEFT JOIN inaturalist.manual_name_additions
+        on (cast(md5(taxa.name) as uuid) = manual_name_additions.md5_scientificname)
+    where taxa.name <> 'Not assigned'
+);
+ALTER TABLE inaturalist.taxa_with_vernacular ADD PRIMARY KEY (taxon_id);
+COMMIT;
+
+/*
+           ********** Create enriched table with ancestry tags *********
+
+Join each record to all of its ancestor records and aggregate ancestor titles into the
+tags along with json strings from the enriched data above.
+    + expand each ancestry string into an array
+    + get the taxa record for each value of the array
+    + aggregate back to the original taxon level, with tags for ancestor names (titles)
+*/
+DROP table if exists inaturalist.taxa_enriched;
+create table inaturalist.taxa_enriched as
 (
     select
         child.taxon_id,
-        array_agg(ancestors.name) as ancestor_names
+        child.title,
+        jsonb_path_query_array(
+            (
+                coalesce(child.inaturalist_name_tag, to_jsonb(array[]::openverse_tag[]))
+                || coalesce(jsonb_agg(DISTINCT
+                    cast((ancestors.title,'inaturalist') as openverse_tag))
+                    FILTER (where ancestors.title is not null),
+                    to_jsonb(array[]::openverse_tag[]))
+                || coalesce(child.tags_nonenglish_vernacular,
+                    to_jsonb(array[]::openverse_tag[]))
+            ),
+            '$[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]')
+        as tags
     from
-        inaturalist.taxa ancestors,
-        inaturalist.taxa child
-    where ancestors.taxon_id = ANY (string_to_array(child.ancestry, '/')::int[])
-        and ancestors.rank not in ('stateofmatter','epifamily','zoosection')
-    group by child.taxon_id
-)
-UPDATE inaturalist.taxa
-SET ancestor_names = aggregated.ancestor_names
-from aggregated
-where taxa.taxon_id = aggregated.taxon_id;
+        inaturalist.taxa_with_vernacular child
+        left join inaturalist.taxa_with_vernacular ancestors
+        on (ancestors.taxon_id = ANY (string_to_array(child.ancestry, '/')::int[]))
+    group by child.taxon_id, child.title,
+        child.inaturalist_name_tag, child.tags_nonenglish_vernacular
```

Comment:
I'm having a particularly hard time grokking all these tables 😅 Would it be possible/reasonable to add an ERD as part of the documentation describing the relationships? Or some other way of describing these relationships?